mlflow/system_metrics/system_metrics_monitor.py

```diff
@@ -60,8 +69,26 @@ def __init__(self, run_id, sampling_interval=10, samples_before_logging=1):
         self.mlflow_logger = BatchMetricsLogger(self._run_id)
         self._shutdown_event = threading.Event()
         self._process = None
-        self._logging_step = 0
         self._metrics_prefix = "system/"
+        self._logging_step = self._get_next_logging_step(run_id) if resume_logging else 0
+
+    def _get_next_logging_step(self, run_id):
+        from mlflow.tracking.client import MlflowClient
+
+        client = MlflowClient()
+        try:
+            run = client.get_run(run_id)
+        except Exception:
+            return 0
+        system_metric_name = None
+        for metric_name in run.data.metrics.keys():
+            if metric_name.startswith(self._metrics_prefix):
+                system_metric_name = metric_name
+                break
+        if system_metric_name is None:
+            return 0
```

Comment:
We can use `next` here:

```python
    system_metric_name = next(
        (m for m in run.data.metrics if m.startswith(self._metrics_prefix)), None
    )
    if system_metric_name is None:
        return 0
```

Reply:
I think the original code is easier to read? 

`(m for m in run.data.metrics if m.startswith(self._metrics_prefix)), None` inlines multiple things.

Reply:
agree